app/src/main/kotlin/nl/eduid/screens/personalinfo/PersonalInfoRepository.kt

```diff
@@ -39,6 +40,30 @@ class PersonalInfoRepository(private val eduIdApi: EduIdApi) {
         null
     }
 
+    suspend fun getErringUserDetails(): UserDetails? = try {
+        val response = eduIdApi.getUserDetails()
+        if (response.isSuccessful) {
+            response.body()
+        } else {
+            if (response.code() == java.net.HttpURLConnection.HTTP_UNAUTHORIZED) {
+                throw UnauthorizedException("Unauthorized getUserDetails to mobile/api/sp/me", null)
```

Comment:
Here you are putting the call path into the error. I think this is unnecessary, since it could change later, and then we forget to update it here.